jetbrains-ultimate/src/software/aws/toolkits/jetbrains/services/caws/DevEnvStatusWatcher.kt

```diff
@@ -0,0 +1,109 @@
+// Copyright 2023 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package software.aws.toolkits.jetbrains.services.caws
+
+import com.intellij.openapi.components.service
+import com.intellij.openapi.project.Project
+import com.intellij.openapi.startup.StartupActivity
+import com.intellij.openapi.ui.MessageDialogBuilder
+import com.jetbrains.rdserver.unattendedHost.UnattendedStatusUtil
+import kotlinx.coroutines.delay
+import kotlinx.coroutines.launch
+import kotlinx.coroutines.runBlocking
+import kotlinx.coroutines.withContext
+import software.amazon.awssdk.services.codecatalyst.CodeCatalystClient
+import software.aws.toolkits.core.utils.getLogger
+import software.aws.toolkits.jetbrains.core.awsClient
+import software.aws.toolkits.jetbrains.core.coroutines.getCoroutineBgContext
+import software.aws.toolkits.jetbrains.core.coroutines.getCoroutineUiContext
+import software.aws.toolkits.jetbrains.core.coroutines.projectCoroutineScope
+import software.aws.toolkits.jetbrains.core.credentials.sono.SonoCredentialManager
+import software.aws.toolkits.jetbrains.services.caws.envclient.CawsEnvironmentClient
+import software.aws.toolkits.jetbrains.services.caws.envclient.models.UpdateActivityRequest
+import software.aws.toolkits.jetbrains.utils.isRunningOnRemoteBackend
+import software.aws.toolkits.jetbrains.utils.notifyError
+import software.aws.toolkits.resources.message
+
+class DevEnvStatusWatcher : StartupActivity {
+
+    companion object {
+        fun getInstance(project: Project) = project.service<DevEnvStatusWatcher>()
+        private val LOG = getLogger<DevEnvStatusWatcher>()
+    }
+
+    override fun runActivity(project: Project) {
+        if (!isRunningOnRemoteBackend()) {
```

Comment:
this isn't quite right -- we only want to run in a CodeCatalyst environment

Reply:
Got it, Adding a check for the CodeCatalyst env id  constant